lib/matplotlib/ticker.py

```diff
@@ -1024,7 +1024,7 @@ def format_data(self, value):
 
     def format_data_short(self, value):
         # docstring inherited
-        return '%-12g' % value
+        return ('%-12g' % value).rstrip()
```

Comment:
Sorry for the post merge comment here, but why not just:

```python
    return f"{value:g}"
```

The 12 in the format is adding padding that we are then just stripping away...